source/Octostache/Templates/Functions/TextSubstringFunction.cs

```diff
@@ -0,0 +1,22 @@
+using System.Linq;
+
+namespace Octostache.Templates.Functions
+{
+    internal static class TextSubstringFunction
+    {
+        public static string Substring(string argument, string[] options)
+        {
+            if (argument == null ||
+                options.Length == 0 ||
+                options.Any(o => !int.TryParse(o, out int _)) ||
+                options.Any(o => int.Parse(o) < 0) ||
+                (options.Length == 1 && int.Parse(options[0]) > argument.Length) ||
+                (options.Length == 2 && (int.Parse(options[0]) + int.Parse(options[1]) > argument.Length)))
```

Comment:
If options length == 3 this part of the validation is skipped.
```suggestion
                (options.Length >= 2 && (int.Parse(options[0]) + int.Parse(options[1]) > argument.Length)))
```

Reply:
On second thought it would be more consistent to return null if `options.Length` isn't 1 or 2.